framework/wazuh/core/cluster/worker.py

```diff
@@ -379,34 +472,29 @@ def remove_agent_file_type(agent_files: List[str]):
         logger.debug("Agents to remove: {}".format(', '.join(agent_ids_list)))
         # the agents must be removed in groups of 997: 999 is the limit of SQL variables per query. Limit and offset are
         # always included in the SQL query, so that leaves 997 variables as limit.
-        for agents_ids_sublist in itertools.zip_longest(*itertools.repeat(iter(agent_ids_list), 997), fillvalue='0'):
+        for agents_ids_sublist in itertools.zip_longest(*itertools.repeat(iter(agent_ids_list), 500), fillvalue='0'):
             agents_ids_sublist = list(filter(lambda x: x != '0', agents_ids_sublist))
             # Get info from DB
             agent_info = Agent.get_agents_overview(q=",".join(["id={}".format(i) for i in agents_ids_sublist]),
                                                    select=['ip', 'id', 'name'], limit=None)['items']
             logger.debug2("Removing files from agents {}".format(', '.join(agents_ids_sublist)))
 
-            files_to_remove = ['{ossec_path}/queue/agent-info/{name}-{ip}',
-                               '{ossec_path}/queue/rootcheck/({name}) {ip}->rootcheck',
+            files_to_remove = ['{ossec_path}/queue/rootcheck/({name}) {ip}->rootcheck',
                                '{ossec_path}/queue/diff/{name}', '{ossec_path}/queue/agent-groups/{id}',
                                '{ossec_path}/queue/rids/{id}',
                                '{ossec_path}/var/db/agents/{name}-{id}.db']
             remove_agent_file_type(files_to_remove)
 
             logger.debug2("Removing agent group assigments from database")
             # remove agent from groups
-            db_global = glob.glob(common.database_path_global)
-            if not db_global:
-                raise WazuhInternalError(1600)
+            wdb_conn = WazuhDBConnection()
 
-            conn = Connection(db_global[0])
-            agent_ids_db = {'id_agent{}'.format(i): int(i) for i in agents_ids_sublist}
-            conn.execute('delete from belongs where {}'.format(
-                ' or '.join(['id_agent = :{}'.format(i) for i in agent_ids_db.keys()])), agent_ids_db)
-            conn.commit()
+            query_to_execute = 'global sql delete from belongs where {}'.format(' or '.join([
+                'id_agent = {}'.format(agent_id) for agent_id in agents_ids_sublist
+            ]))
+            wdb_conn.run_wdb_command(query_to_execute)
 
             # Tell wazuhbd to delete agent database
-            wdb_conn = WazuhDBConnection()
             wdb_conn.delete_agents_db(agents_ids_sublist)
 
```

Comment:
Add a exception if response is not OK

Reply:
Removed because agents are automatically deleted when the `client.keys` file is updated.